src/http/web_socket.cr

```diff
@@ -60,7 +60,7 @@ class HTTP::WebSocket
   def on_binary(&@on_binary : Bytes ->)
   end
 
-  def on_close(&@on_close : Int32, String ->)
+  def on_close(&@on_close : CloseCode, String ->)
```

Comment:
I'm not sure that giving application code an invalid enum is the best approach...

Reply:
What do you mean by _invalid_? It's more valid than passing an arbitrary `Int` and it was already mentioned in https://github.com/crystal-lang/crystal/pull/8981#issuecomment-607292537.

Reply:
I think it's fine.

Reply:
Enums eventually won't be allowed to have invalid members. This shouldn't be made API.

Reply:
They won't, unless they are marked with a special attribute that allows them that. I think it's convenient for cases like HTTP::Status and this, where there are some pre-defined codes but also user-defined values.